app/src/main/java/com/keylesspalace/tusky/worker/NotificationWorker.kt

```diff
@@ -17,27 +17,36 @@
 
 package com.keylesspalace.tusky.worker
 
+import android.app.Notification
 import android.content.Context
-import androidx.work.Worker
+import androidx.work.CoroutineWorker
+import androidx.work.ForegroundInfo
 import androidx.work.WorkerParameters
+import com.keylesspalace.tusky.R
 import com.keylesspalace.tusky.components.notifications.NotificationFetcher
+import com.keylesspalace.tusky.components.notifications.NotificationHelper
+import com.keylesspalace.tusky.components.notifications.NotificationHelper.NOTIFICATION_ID_FETCH_NOTIFICATION
 import javax.inject.Inject
 
 /** Fetch and show new notifications. */
 class NotificationWorker(
     appContext: Context,
     params: WorkerParameters,
     private val notificationsFetcher: NotificationFetcher
-) : Worker(appContext, params) {
-    override fun doWork(): Result {
+) : CoroutineWorker(appContext, params) {
+    val notification: Notification = NotificationHelper.createWorkerNotification(applicationContext, R.string.notification_notification_worker)
+
+    override suspend fun doWork(): Result {
         notificationsFetcher.fetchAndShow()
         return Result.success()
     }
 
+    override suspend fun getForegroundInfo() = ForegroundInfo(NOTIFICATION_ID_FETCH_NOTIFICATION, notification)
```

Comment:
It works on Android 12 because `getForegroundInfo()` isn't called on Android 12, there `getForegroundInfoAsync()` is called instead, https://developer.android.com/reference/kotlin/androidx/work/CoroutineWorker#getForegroundInfo()